src/attribution/onLCP.ts

```diff
@@ -22,19 +22,22 @@ import {
   LCPAttribution,
   LCPMetric,
   LCPMetricWithAttribution,
-  LCPReportCallback,
-  LCPReportCallbackWithAttribution,
   ReportOpts,
 } from '../types.js';
 
-const attributeLCP = (metric: LCPMetric) => {
-  if (metric.entries.length) {
-    const navigationEntry = getNavigationEntry();
-
-    if (navigationEntry) {
-      const responseStart = navigationEntry.responseStart;
-      if (isInvalidTimestamp(responseStart)) return;
+const attributeLCP = (metric: LCPMetric): LCPMetricWithAttribution => {
+  // Use a default object if no other attribution has been set.
+  let attribution: LCPAttribution = {
+    timeToFirstByte: 0,
+    resourceLoadDelay: 0,
+    resourceLoadDuration: 0,
+    elementRenderDelay: metric.value,
+  };
 
+  const navigationEntry = getNavigationEntry();
+  if (metric.entries.length && navigationEntry) {
+    const responseStart = navigationEntry.responseStart;
+    if (!isInvalidTimestamp(responseStart)) {
```

Comment:
almost tempted to break to label here :) Moving the `getNavigationEntry` out seems ok since I assume the `!metric.entries.length` case is very rare.

Reply:
> Now that we're adding `!` before `isInvalidTimestamp()` it reads awkwardly, and TBH I think it could be refactored into the `getNavigationEntry()` function since it's always used in that context.
> 
> Did you configure your fork to let me push to it in the PR? (Or could you just give me access?) I'd be happy to experiment with some options in this PR.

Makes sense. I just enabled pushes to this PR (also you should add me to `web-vitals` :)

Reply:
Added! (I didn't realize you weren't already there.)